### ISSUE TITLE
jasonxiale - `BunniPrice.getBunniTokenPrice` calculates price incorrectly

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-11-olympus-judging#125]
jasonxiale

high

# `BunniPrice.getBunniTokenPrice` calculates price incorrectly

## Summary
`BunniPrice.getBunniTokenPrice` is supposed to calculate the `bunniToken_` (representing a Uniswap V3 pool) in USD, but current implementation, it calculates the total value instead of price
## Vulnerability Detail
In current implementation of [BunniPrice.getBunniTokenPrice](https://github.com/sherlock-audit/2023-11-olympus/blob/9c8df76dc9820b4c6605d2e1e6d87dcfa9e50070/bophades/src/modules/PRICE/submodules/feeds/BunniPrice.sol#L110-L166), the function:
1. check if [the parameters is correct and valiadte reserve](https://github.com/sherlock-audit/2023-11-olympus/blob/9c8df76dc9820b4c6605d2e1e6d87dcfa9e50070/bophades/src/modules/PRICE/submodules/feeds/BunniPrice.sol#L116-L152)
1. Validate reserves by calling [_validateReserves](https://github.com/sherlock-audit/2023-11-olympus/blob/9c8df76dc9820b4c6605d2e1e6d87dcfa9e50070/bophades/src/modules/PRICE/submodules/feeds/BunniPrice.sol#L155-L160)
1. calling [_getTotalValue](https://github.com/sherlock-audit/2023-11-olympus/blob/9c8df76dc9820b4c6605d2e1e6d87dcfa9e50070/bophades/src/modules/PRICE/submodules/feeds/BunniPrice.sol#L163)

And n [function _getTotalValue](https://github.com/sherlock-audit/2023-11-olympus/blob/9c8df76dc9820b4c6605d2e1e6d87dcfa9e50070/bophades/src/modules/PRICE/submodules/feeds/BunniPrice.sol#L215-L233), the function
1. get reserve by calling [_getBunniReserves](https://github.com/sherlock-audit/2023-11-olympus/blob/9c8df76dc9820b4c6605d2e1e6d87dcfa9e50070/bophades/src/modules/PRICE/submodules/feeds/BunniPrice.sol#L192-L207)
1. calculate the total value in [BunniPrice.sol#L228-L232](https://github.com/sherlock-audit/2023-11-olympus/blob/9c8df76dc9820b4c6605d2e1e6d87dcfa9e50070/bophades/src/modules/PRICE/submodules/feeds/BunniPrice.sol#L228-L232)
```solidity
215     function _getTotalValue(
216         BunniToken token_,
217         BunniLens lens_,
218         uint8 outputDecimals_
219     ) internal view returns (uint256) {
...
227         // Determine the value of each reserve token in USD
228         uint256 totalValue;
229         totalValue += _PRICE().getPrice(token0).mulDiv(reserve0, outputScale);
230         totalValue += _PRICE().getPrice(token1).mulDiv(reserve1, outputScale);
231 
232         return totalValue; <<<--- calculate total value
233     }

```
**To sum up, the function calculates the total value of bunnie pool instead of `bunniToken_` price**

## Impact
`BunniPrice.getBunniTokenPrice` is supposed to calculate the `bunniToken_` (representing a Uniswap V3 pool) in USD, but current implementation, it calculates the total value instead of price
## Code Snippet
https://github.com/sherlock-audit/2023-11-olympus/blob/9c8df76dc9820b4c6605d2e1e6d87dcfa9e50070/bophades/src/modules/PRICE/submodules/feeds/BunniPrice.sol#L110-L166
## Tool used

Manual Review

## Recommendation

Duplicate of #198
